planner/core/find_best_task.go

```diff
@@ -249,6 +249,14 @@ func (p *baseLogicalPlan) enumeratePhysicalPlans4Task(physicalPlans []PhysicalPl
 			}
 		}
 
+		// convert to root task as needed to satisfy property
```

Comment:
It's no need to do this check in this PR. Actually it's none business of projection pushdown.

Reply:
we should add this, because all plan would be pushdown, so there would be not root task.